projects/common/src/telemetry/user-telemetry-impl.service.ts

```diff
@@ -9,7 +9,7 @@ import { UserTelemetryService } from './user-telemetry.service';
 export class UserTelemetryImplService extends UserTelemetryService {
   private telemetryProviders: UserTelemetryInternalConfig[] = [];
 
-  public constructor(private readonly injector: Injector, private readonly router: Router) {
+  public constructor(private readonly injector: Injector, @Optional() private readonly router?: Router) {
```

Comment:
So, if I don't make Router optional then I get test errors wherever the atomic component (that uses htTrack) is used. Alternatively, I will have to provide Router Mock service everywhere.

Reply:
But shouldn't we be mocking the component in those tests? Why are they use the real track component?